rtlive/sources/data_be.py

```diff
@@ -1,39 +1,225 @@
 
-from . import ourworldindata
 from .. import preprocessing
 
 _log = logging.getLogger(__file__)
 
+# From https://en.wikipedia.org/wiki/Provinces_of_Belgium
+BE_REGION_NAMES = {
+    'all': 'Belgium',
+    'Flanders': 'Flanders',
+    'Wallonia': 'Wallonia',
+    'Brussels': 'Brussels',
+    'Antwerpen': 'Antwerp',
+    'Limburg': 'Limburg',
+    'OostVlaanderen': 'East Flanders',
+    'VlaamsBrabant': 'Flemish Brabant',
+    'WestVlaanderen': 'West Flanders',
+    'Hainaut': 'Hainaut',
+    'Liège': 'Liège',
+    'Luxembourg': 'Luxembourg',
+    'Namur': 'Namur',
+    'BrabantWallon': 'Walloon Brabant',
+}
+
+# Province and region codes
+# [ISO 3166-2:BE](https://en.wikipedia.org/wiki/ISO_3166-2:BE#Provinces) has no english codes
+BE_REGION_ABBR = {
+    'all': 'all',
+    'Flanders': 'FLA',
+    'Wallonia': 'WAL',
+    'Brussels': 'BRU',
+    'Antwerpen': 'ANT',
+    'Limburg': 'LIM',
+    'OostVlaanderen': 'EFL',
+    'VlaamsBrabant': 'FBR',
+    'WestVlaanderen': 'WFL',
+    'Hainaut': 'HAI',
+    'Liège': 'LIE',
+    'Luxembourg': 'LUX',
+    'Namur': 'NAM',
+    'BrabantWallon': 'WBR',   
+}
 
-def forecast_BE(df: pandas.DataFrame):
-    """ Applies testcount interpolation/extrapolation.
+BE_REGION_CODES = {
+    v : k
+    for k, v in BE_REGION_NAMES.items()
+}
 
-    Currently this assumes the OWID data, which only has an "all" region.
-    In the future, this should be replaced with more fine graned data loading!
+# Source: https://www.ibz.rrn.fgov.be/fileadmin/user_upload/fr/pop/statistiques/population-bevolking-20200101.pdf
+BE_REGION_POPULATION = {
+    'all':           11_476_279, # Belgium
+    'Flanders':       6_623_505,
+    'Wallonia':       3_641_748,
+    'Brussels':       1_211_026,
+    'Antwerpen':      1_867_366,
+    'Limburg':          876_785,
+    'OostVlaanderen': 1_524_077,
+    'VlaamsBrabant':  1_155_148,
+    'WestVlaanderen': 1_200_129,
+    'Hainaut':        1_345_270,
+    'Liège':          1_108_481,
+    'Luxembourg':       286_571,
+    'Namur':            495_474,
+    'BrabantWallon':    405_952
+}
+
+def get_data_BE(run_date) -> pandas.DataFrame:
+    """
+    Retrieve daily (run_date) regions and append national data (key 'all') to it
+    Parameters
+    ----------
+    run_date : pandas.Timestamp
+        date for which the data shall be downloaded
+    
+    Returns
+    -------
+    df : pandas.DataFrame
+        table with columns as required by rtlive/data.py API
+    """
+    
+    def redistribute(g, col):
+        gdata = g.groupby('REGION')[col].sum()
+        gdata.loc['Brussels'] += gdata.loc['Nan'] * (gdata.loc['Brussels']/(gdata.loc['Brussels'] + gdata.loc['Flanders'] + gdata.loc['Wallonia']))
+        gdata.loc['Flanders'] += gdata.loc['Nan'] * (gdata.loc['Flanders']/(gdata.loc['Brussels'] + gdata.loc['Flanders'] + gdata.loc['Wallonia']))
+        gdata.loc['Wallonia'] += gdata.loc['Nan'] * (gdata.loc['Wallonia']/(gdata.loc['Brussels'] + gdata.loc['Flanders'] + gdata.loc['Wallonia']))
```

Comment:
Also here we use the short codes on the index, so you'll have to replace them. You can use the `BE_REGION_CODES` lookup for that.